src/flake8_requirements/checker.py

```diff
@@ -73,6 +73,8 @@ def joinlines(lines):
     joined_line = ""
     for line in map(lambda x: x.strip(), lines):
         if not line or line.startswith("#"):
+            yield joined_line
+            joined_line = ""
             continue
         if line.endswith("\\"):
```

Comment:
I think if-blocks on `75` and `79` lines could be swapped which would fix the issue, the same as [pip does](https://github.com/pypa/pip/blob/025e99ec61039649ebdad079244379600ccdfe8a/src/pip/_internal/req/req_file.py#L153-L154).

Reply:
Yes, you are righ, that would make flake8-requirements work like pip. But there is a documentation for requirements.txt file which explicitly says "comments are stripped *before* line continuations are processed". So either documentation is wrong, or pip implementation....

Reply:
I just saw the talk on pypa/pip, so when the issue pypa/pip/issues/7728 will be answered, I will follow accordingly :)